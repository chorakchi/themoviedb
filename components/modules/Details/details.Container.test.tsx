// @flow

import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { Details } from "./details.Container";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import { mount } from "enzyme";
import { createStore } from "redux";
import { INITIAL_STATE as reducer } from "./details.reducer";

describe("details unit test", () => {
  it("should render correctly", () => {
    const mockStore = configureMockStore([thunk]);
    const store = mockStore({
      details: reducer,
    });
    const subject = renderer
      .create(
        <Provider store={store}>
          <Details />
        </Provider>
      )
      .toJSON();
    expect(subject).toMatchSnapshot();
  });

  // const getWrapper = (mockStore = createStore(() => reducer, reducer)) =>
  //   mount(
  //     <Provider store={mockStore}>
  //       <Details />
  //     </Provider>
  //   );

  // it("should dispatch run  correctly ", () => {
  //   const mockStore = createStore(() => reducer, reducer);
  //   mockStore.dispatch = jest.fn();
  //   const wrapper = getWrapper(mockStore);
  //   wrapper.find("#filtersContainer").get(0);
  //   // .props.onChange("mockCallback");
  //   expect(mockStore.dispatch).toHaveBeenCalledTimes(1);
  // });
});
