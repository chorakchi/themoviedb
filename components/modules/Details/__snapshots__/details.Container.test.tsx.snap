// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`details unit test should render correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
/>
`;
